changelogs/fragments/6571.yml

```diff
@@ -0,0 +1,2 @@
+refactor:
+- discover data selector enhancement and refactoring ([#6571](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6571))
```

Comment:
Can we verify it's just Discover? I think it's any thing that uses it the data source selector right but just doesn't pass it.

Reply:
For now it's just discover, but since we have already aligned on having data selector as its name to distinguish from MDS selector I think we can remove 'discover' as it can be used outside of discover.

Reply:
gotcha.

but also if some 3rd party plugin referenced this selector technically it would be modified as well right?

Reply:
yea that is correct.

Reply:
sounds good to me! thanks!